packages/daemon/src/daemon-node-powers.js

```diff
@@ -504,6 +504,12 @@ export const makeDaemonicPersistencePowers = (
     await Promise.all([statePathP, cachePathP, ephemeralStatePathP]);
   };
 
+  const isRootInitialized = async () => {
+    const noncePath = filePowers.joinPath(locator.statePath, 'nonce');
+    const nonce = await filePowers.maybeReadFileText(noncePath);
+    return nonce !== undefined;
+  };
+
```

Comment:
We could avoid attempting to read the root nonce file twice by merging this into `provideRootNonce` by making the latter return `{ value: string, isNewlyCreated: boolean }` (alternatively `wasRestored: boolean`).